source/crud/tests/unified/aggregate-write-readPreference.yml

```diff
@@ -78,12 +75,6 @@ tests:
                 $readPreference: *readPreference
                 readConcern: *readConcern
                 writeConcern: *writeConcern
-    outcome: &outcome
```

Comment:
Was it also necessary to remove the `outcome` assertions? I didn't see that mentioned in [DRIVERS-2914](https://jira.mongodb.org/browse/DRIVERS-2914) (only `initialData`), and you didn't remove this assertion from `db-aggregate-write-readPreference.yml` below.

I'll concede that this isn't essential to the test. The `commandStartedEvent` assertions for `$readPreference` is the most relevant bit.

Reply:
I've ran into issues where the outcome assertion failed with no documents being found, but they were there when I checked. Step-debugging and introducing a tiny delay fixed the issue, so I decided to remove that element of flakiness from the test, as the real test is the assertion for `$readPreference`, and we're testing server behaviour beyond that.

Not removing the assertion from `db-aggregate-write-readPreference` was an oversight and I've now removed it from that test as well for consistency.